commandapi-core/src/main/java/dev/jorel/commandapi/AbstractCommandAPICommand.java

```diff
@@ -83,7 +84,41 @@ public Impl withArguments(List<Argument> args) {
 	 */
 	@SafeVarargs
 	public final Impl withArguments(Argument... args) {
-		this.args.addAll(Arrays.asList(args));
+		this.arguments.addAll(Arrays.asList(args));
+		return instance();
+	}
+
+	/**
+	 * Appends the optional arguments to the current command builder.
+	 *
+	 * This also calls {@link AbstractArgument#setOptional(boolean)} on each argument to make sure they are optional
+	 *
+	 * @param args A <code>List</code> that represents the arguments that this
+	 *             command can accept
+	 * @return this command builder
+	 */
+	public Impl withOptionalArguments(List<Argument> args) {
+		args.forEach(argument -> {
+			argument.setOptional(true);
+			this.arguments.add(argument);
+		});
```

Comment:
Use a for-each loop instead of the `forEach` method:

```java
for(Argument argument : args) {
	argument.setOptional(true);
	this.arguments.add(argument);
}
```